### ISSUE TITLE
Cosmos checkstyle issues

### ISSUE THREAD

**Issue by srnagar** [Azure/azure-sdk-for-java#8681]
There are some checkstyle issues in Cosmos reported in [report-generation pipeline ](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=276241&view=results)

```
src/main/java/com/azure/cosmos/CosmosDatabase.java(243,17): error : src/main/java/com/azure/cosmos/CosmosDatabase.java:[243,17] (extension) ThrowFromClientLogger: Directly throwing an exception is disallowed. Must throw through 'ClientLogger' API, either of 'logger.logExceptionAsError' or 'logger.logExceptionAsWarning' where 'logger' 
src/main/java/com/azure/cosmos/CosmosDatabase.java(435,17): error : src/main/java/com/azure/cosmos/CosmosDatabase.java:[435,17] (extension) ThrowFromClientLogger: Directly throwing an exception is disallowed. Must throw through 'ClientLogger' API, either of 'logger.logExceptionAsError' or 'logger.logExceptionAsWarning' where 'logger' is type of Clie 
src/main/java/com/azure/cosmos/CosmosDatabase.java(473,17): error : src/main/java/com/azure/cosmos/CosmosDatabase.java:[473,17] (extension) ThrowFromClientLogger: Directly throwing an exception is disallowed. Must throw through 'ClientLogger' API, either of 'logger.logExceptionAsError' or 'logger.logExceptionAsWarning' where 'logger' is 
src/main/java/com/azure/cosmos/DatabaseAccount.java(197,21): error : src/main/java/com/azure/cosmos/DatabaseAccount.java:[197,21] (extension) ThrowFromClientLogger: Directly throwing an exception is disallowed. Must throw through 'ClientLogger' API, either of 'logger.lo 
src/main/java/com/azure/cosmos/CosmosAsyncItemResponse.java(9,8): error : src/main/java/com/azure/cosmos/CosmosAsyncItemResponse.java:[9,8] (imports) UnusedImports: Unused import - org.apache.commons.lang3.StringUtils. 
src/main/java/com/azure/cosmos/implementation/ReplicatedResourceClientUtils.java(18,62): error : src/main/java/com/azure/cosmos/implementation/ReplicatedResourceClientUtils.java:[18,62] (whitespace) OperatorWrap: '||' should be on a new line. 
src/main/java/com/azure/cosmos/implementation/ReplicatedResourceClientUtils.java(30,55): error : src/main/java/com/azure/cosmos/implementation/ReplicatedResourceClientUtils.java:[30,55] (whitespace) Operat 
src/main/java/com/azure/cosmos/implementation/ReplicatedResourceClientUtils.java(32,62): error : src/main/java/com/azure/cosmos/implementation/ReplicatedResourceClientUtils.java:[32,62] (whitespace) OperatorWrap: '||' should be on a new line. 
src/main/java/com/azure/cosmos/implementation/ReplicatedResourceClientUtils.java(33,57): error : src/main/java/com/azure/cosmos/implementation/ReplicatedResourceClientUtils.java:[33,57] (whitespace) OperatorWrap: 
src/main/java/com/azure/cosmos/implementation/ReplicatedResourceClientUtils.java(34,55): error : src/main/java/com/azure/cosmos/implementation/ReplicatedResourceClientUtils.java:[34,55] (whitespace) OperatorWrap: '||' should be on a new line. 
Build failed. 
```

**Comment by mbhaskar** [comment]
@srnagar failsOnError/failsOnViolation is false for cosmos checkstyle. Why does the build fail for you?

**Comment by srnagar** [comment]
@mbhaskar for report generation, we have a configuration in `pom.client.xml` which has failsOnError and failsOnViolation set to true. The configuration in cosmos pom.xml only applies to `install` or `package` goals. Report generation uses `site:site site:stage` goal and the plugin is configured differently for this goal.
```xml
<reporting>
    <plugins>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-checkstyle-plugin</artifactId>
        <version>3.1.0</version> <!-- {x-version-update;org.apache.maven.plugins:maven-checkstyle-plugin;external_dependency} -->
        <configuration>
          <configLocation>eng/code-quality-reports/src/main/resources/checkstyle/checkstyle.xml</configLocation>
          <suppressionsLocation>eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml
          </suppressionsLocation>
          <headerLocation>eng/code-quality-reports/src/main/resources/checkstyle/java.header</headerLocation>
          <propertyExpansion>samedir=</propertyExpansion>
          <encoding>UTF-8</encoding>
          <consoleOutput>true</consoleOutput>
          <includeTestSourceDirectory>true</includeTestSourceDirectory>
          <linkXRef>true</linkXRef>
          <failsOnError>true</failsOnError>
          <failOnViolation>true</failOnViolation>
        </configuration>
        <reportSets>
          <reportSet>
            <id>non-aggregate</id>
            <reports>
              <report>checkstyle</report>
            </reports>
          </reportSet>
          <reportSet>
            <id>aggregate</id>
            <inherited>false</inherited>
            <reports>
              <report>checkstyle-aggregate</report>
            </reports>
          </reportSet>
        </reportSets>
      </plugin>

....
<remaining part of the reporting section is skipped>
```

**Comment by srnagar** [comment]
Fixed in #8789
